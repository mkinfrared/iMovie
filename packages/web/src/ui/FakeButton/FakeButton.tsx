import React, { KeyboardEvent, useCallback } from "react";

import { FakeButtonProps } from "./FakeButton.type";

const FakeButton = ({ children, onClick, className }: FakeButtonProps) => {
  const handleKeyDown = useCallback(
    (event: KeyboardEvent) => {
      if (event.key === "Enter") {
        onClick();
      }
    },
    [onClick]
  );

  return (
    <div
      className={className}
      onClick={onClick}
      onKeyDown={handleKeyDown}
      role="button"
      tabIndex={0}
      data-testid="fakeButton"
    >
      {children}
    </div>
  );
};

export default React.memo(FakeButton);
