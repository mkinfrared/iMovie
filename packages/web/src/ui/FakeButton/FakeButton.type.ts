export interface FakeButtonProps {
  children?: React.ReactNode;
  className?: string;
  onClick: () => void;
}
