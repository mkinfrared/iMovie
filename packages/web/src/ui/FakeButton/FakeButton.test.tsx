import { fireEvent, render } from "@testing-library/react";
import React, { ReactElement } from "react";

import FakeButton from "./FakeButton";

describe("<FakeButton />", () => {
  const onClick = jest.fn();

  let Component: ReactElement;

  beforeEach(() => {
    Component = <FakeButton onClick={onClick} />;
  });

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match a snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should call 'onClick' when 'Enter' is pressed", () => {
    const { getByTestId } = render(Component);
    const fakeButton = getByTestId("fakeButton");

    fireEvent.keyDown(fakeButton, { key: "Enter" });

    expect(onClick).toHaveBeenCalled();
  });

  it("should not call 'onClick' when key pressed is not 'Enter'", () => {
    const { getByTestId } = render(Component);
    const fakeButton = getByTestId("fakeButton");

    fireEvent.keyDown(fakeButton, { key: "Alt" });

    expect(onClick).toHaveBeenCalled();
  });
});
