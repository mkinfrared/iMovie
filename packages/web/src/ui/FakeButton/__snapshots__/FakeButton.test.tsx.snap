// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FakeButton /> should match a snapshot 1`] = `
<div>
  <div
    data-testid="fakeButton"
    role="button"
    tabindex="0"
  />
</div>
`;
